core/src/crypto/DeterministicPublicKey.cpp

```diff
@@ -88,6 +89,15 @@ namespace ledger {
             return  std::vector<uint8_t>(keccak.end() - 20, keccak.end());
         }
 
+        std::vector<uint8_t> DeterministicPublicKey::getPublicKeyBlake2b(bool isED25519) const {
+            auto key = _key;
+            if (!isED25519) {
+                //Remove 0x02
+                key.erase(key.begin());
```

Comment:
This is a O(n) operation that could have been prevented if you made `BLAKE::blake2b` use a slice / iterator instead of using a `std::vector`.

Reply:
`auto key = _key;` is O(n) as well.

Reply:
Nice catch, this was my "blind" test phase but forgot to remove it, I was willing to have an offset on `BLAKE::blake2b` that would solve the issue !